herd/AArch64Sem.ml

```diff
@@ -1901,9 +1900,8 @@ module Make
       let simd_str = do_simd_str Annot.N
       let simd_stlr = do_simd_str Annot.L
 
-      let simd_str_p sz rs rd k ii =
-        read_reg_ord rs ii >>|
-        read_reg_neon true rd ii >>= fun (addr, v) ->
+      let simd_str_p sz addr rd rs k ii =
+        read_reg_neon true rd ii >>= fun v ->
```

Comment:
Check my comment below but to retain the iico_* relations as before, then you could do:
```suggestion
      let simd_str_p sz ma rd rs k ii =
        ma >>| 
        read_reg_neon true rd ii >>= fun (addr, v) ->
```